src/Orchard.Web/Modules/Orchard.Comments/Drivers/CommentsPartDriver.cs

```diff
@@ -77,6 +77,22 @@ public class CommentsPartDriver : ContentPartDriver<CommentsPart> {
                         return shapeHelper.Parts_Comments_Count(
                             CommentCount: part.CommentsCount);
                     }),
+                ContentShape("Parts_Comments_Count_Detail",
+                    () => {
+                        if (part.CommentsShown == false)
+                            return null;
+
+                        return shapeHelper.Parts_Comments_Count_Detail(
+                            CommentCount: part.CommentsCount);
+                    }),

```

Comment:
Why do you need `Parts_Comments_Count_Detail` in addition to `Parts_Comments_Count`? The latter could serve as the "detail" shape by configuring your theme;s Placement.info to not render that shape by default, only for the Detail display type.